plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -27,47 +26,38 @@
 
 import com.openshift.restclient.OpenShiftException;
 import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IPod;
 import com.openshift.restclient.model.IProject;
 import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.IService;
 
 /**
  * Contributes OpenShift 3 specific content to the OpenShift explorer view
  * 
  * @author jeff.cantrill
  */
-public class OpenShiftExplorerContentProvider extends BaseExplorerContentProvider{
+public class OpenShiftExplorerContentProvider extends BaseExplorerContentProvider {
 	
-	private static final String [] groupings = new String [] {
-		ResourceKind.BUILD_CONFIG, 
-		ResourceKind.DEPLOYMENT_CONFIG, 
-		ResourceKind.SERVICE, 
-		ResourceKind.POD,
-		ResourceKind.REPLICATION_CONTROLLER, 
-		ResourceKind.BUILD, 
-		ResourceKind.IMAGE_STREAM, 
-		ResourceKind.ROUTE
-	};
-	
-	private Map<IProject, List<ResourceGrouping>> groupMap = new HashMap<IProject, List<ResourceGrouping>>();
+	private Map<IProject, List<Deployment>> deploymentsByProject = new HashMap<>();
+	private Map<IService, Deployment> deploymentByService = new HashMap<>();
 	
```

Comment:
Yes, this is a trivial issue, but just the same, hashCode() and equals() are very related methods, and if one of them checks or does not check some value for null for some good reason, the other should treat that value in the same way.